packages/flutter_tools/lib/src/base/terminal.dart

```diff
@@ -319,7 +323,7 @@ class AnsiTerminal implements Terminal {
       return false;
     }
     final io.Stdin stdin = _stdio.stdin as io.Stdin;
-    return stdin.lineMode && stdin.echoMode;
+    return !stdin.lineMode && !stdin.echoMode;
```

Comment:
This seems like a huge logic bug, and yet it seems like the only place this was actually called is from tests? Did you observe buggy behavior here, or just observe the code was wrong?

Reply:
The getter did not match the setter.  `set singleCharMode` sets `echoMode` and `lineMode` to `false` if `value` is true, but the getter had reversed this.

AFAICT this getter is not used anywhere except the test added in this PR.